endpoints/openrtb2/sample-requests/invalid-native/context-type-social-incompatible-subtype.json

```diff
@@ -0,0 +1,27 @@
+{
+  "description": "Native bid request. Context type social (2) is incompatible with 'product' subcontext types (30~39). Return error",
+  "mockBidRequest": {
+    "id": "req-id",
+    "site": {
+      "page": "some.page.com"
+    },
+    "tmax": 500,
+    "imp": [
+      {
+        "id": "some-imp",
+        "native": {
+          "request": "{\"context\":2,\"contextsubtype\":31,\"plcmttype\":1,\"assets\":[{\"title\":{\"len\":90}},{\"img\":{\"hmin\":30,\"wmin\":20}},{\"video\":{\"mimes\":[\"video/mp4\"],\"minduration\":5,\"maxduration\":10,\"protocols\":[1]}},{\"data\":{\"type\":2}}]}"
+        },
+        "ext": {
+          "appnexus": {
+            "placementId": 12883451
+          }
+        }
+      }
+    ]
+  },
+  "expectedReturnCode": 400,
+  "expectedErrorMessage": "Invalid request"
```

Comment:
I see the error messages returned in the `validateNativeContextTypes` method to be different that what is expected in these tests. Is the error message getting overridden somewhere to this?? If so, they shouldn't be as the error messages in the `validateNativeContextTypes` are more informative of the problem

Reply:
I noticed this as well but I played with one of the tests a bit and saw that the actual error message returned by `validateNativeContextTypes` has `Invalid Request: ` prepended to it and the test framework is just checking that the error message begins with whatever `expectedErrorMessage` is rather than doing a full string compare.

Reply:
Exactly